components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -0,0 +1,258 @@
+#ifndef SCREAM_SCORPIO_INPUT_HPP
+#define SCREAM_SCORPIO_INPUT_HPP
+
+#include "scream_config.h"
+#include "ekat/ekat_parameter_list.hpp"
+#include "ekat/std_meta/ekat_std_enable_shared_from_this.hpp"
+#include "ekat/mpi/ekat_comm.hpp"
+#include "ekat/std_meta/ekat_std_utils.hpp"
+#include "ekat/ekat_parse_yaml_file.hpp"
+
+#include "share/io/scream_scorpio_interface.hpp"
+
+#include "share/field/field_repository.hpp"
+#include "share/field/field_header.hpp"
+#include "share/field/field.hpp"
+#include "share/field/field_identifier.hpp"
+
+#include "share/grid/grids_manager.hpp"
+
+namespace scream
+{
+
+class AtmosphereInput 
```

Comment:
A few explanatory comments here and for each method would go a long way to help others that encounter this type.